packages/trpc/server/routers/viewer/bookings/confirm.handler.ts

```diff
@@ -425,3 +412,48 @@ export const confirmHandler = async ({ ctx, input }: ConfirmOptions) => {
 
   return { message, status };
 };
+
+const checkIfUserIsAuthorizedToConfirmBooking = async ({
+  eventTypeId,
+  loggedInUserId,
+  teamId,
+  bookingUserId,
+  userRole,
+}: {
+  eventTypeId: number | null;
+  loggedInUserId: number;
+  teamId?: number | null;
+  bookingUserId: number | null;
+  userRole: string;
+}): Promise<void> => {
+  // check system wide admin
+  if (userRole === UserPermissionRole.ADMIN) return;
+
+  // Check if the user is the owner of the event type
+  if (bookingUserId === loggedInUserId) return;
+
+  // Check if user is associated with the event type
+  if (eventTypeId) {
+    const eventType = await prisma.eventType.findUnique({
+      where: {
+        id: eventTypeId,
+        OR: [{ hosts: { some: { userId: loggedInUserId } } }, { users: { some: { id: loggedInUserId } } }],
+      },
+    });
+    if (eventType) return;
+  }
+
+  // Check if the user is an admin/owner of the team the booking belongs to
+  if (teamId) {
+    const membership = await prisma.membership.findFirst({
+      where: {
+        userId: loggedInUserId,
+        teamId: teamId,
+        OR: [{ role: MembershipRole.OWNER }, { role: MembershipRole.ADMIN }],
```

Comment:
We could separate queries here as well, or maybe even use something like

```
role: {
    in: [MembershipRole.OWNER, MembershipRole.ADMIN]
  }
```
where it would potentially not turn this into Seq search